README.md

```diff
@@ -159,6 +159,207 @@ python ./scripts/train_hf_ner.py ./data/annotations/train.spacy ./data/annotatio
+
+
+### Example usage
+
+The `textcat` recipes can be used for binary, multiclass, and multilabel text
+categorization. You can set this by passing the appropriate number of labels in
+the `--labels` parameter; for example, passing a single label turns it into
+binary classification and so on. We will talk about each one in the proceeding
+sections.
+
+#### Binary text categorization
+
+Suppose we want to know if a particular Reddit comment talks about a food
+recipe.  We'll send the text to GPT-3 and provide a prompt that instructs the
+predictions we want. 
+
+```
+From the text below, determine wheter or not it contains a recipe. If it is a 
+recipe, answer "accept." If it is not a recipe, answer "reject."
+
+Your answer should only be in the following format:
+answer: <string>
+reason: <string>
+
+Text:
+```
+
+For binary classification, we want GPT-3 to return "accept" if a given text is a
+food recipe and "reject" otherwise. GPT-3's suggestion is then displayed
+prominently in the UI. We can press the <kbd>ACCEPT</kbd> (check mark) button to
+include the text as a positive example or press the <kbd>REJECT</kbd> (cross
+mark) button if it is a negative example.
+
+
+```sh
+python -m prodigy textcat.openai.correct my_textcat_data data/reddit_r_cooking_sample.jsonl \
+    --labels recipe \
+    --verbose \
+    -F recipes/openai_textcat.py
+```
+
+<img src="https://user-images.githubusercontent.com/12949683/214230166-ee492fe5-04da-4b93-9590-b5ef23ce488d.png" width="600"/>
+
+
+#### Multilabel and multiclass text categorization
+
+Now, suppose we want to classify Reddit comments as a recipe, a feedback, or a
+question. We can write the following prompt: 
+
+```
+Classify the text below to any of the following labels: recipe, feedback, question.
+The task is exclusive, so only choose one label from what I provided.
+
+Your answer should only be in the following format:
+answer: <string>
+reason: <string>
+
+Text:
+```
+
+Then, we can use this recipe to handle multilabel and multiclass cases by
+passing the three labels to the `--labels` parameter. We should also set the
+`--exclusive-classes` flag to render a single-choice UI:
+
+```sh
+python -m prodigy textcat.openai.correct my_textcat_data data/reddit_r_cooking_sample.jsonl \
+    --labels recipe,feedback,question \
+    --verbose \
+    --exclusive-classes \
+    -F recipes/openai_textcat.py
+```
+
+<img src="https://user-images.githubusercontent.com/12949683/214230269-bdacbbc8-8edc-4be5-8334-0b7eaf4712e0.png" width="600" />
+
+#### Writing templates
+
+We write these prompts as a .jinja2 template that can also take in examples for
+few-shot learning. You can create your own [template](templates) and provide it
+to the recipe with the `--prompt-path` or `-p` option.  Additionally, with
+`--examples-path` or `-e` you can set the file path of a .y(a)ml or .json file
+that contains additional examples:
+
+```bash
+python -m prodigy textcat.openai.correct my_textcat_data \
```

Comment:
Maybe use a different name for a dataset here? The binary example already uses `my_textcat_data`, it'd probably be best that single-label and multi-label annotations have their own dataset.